### PR TITLE
Refactoring support to control volume of player android

### PR DESCRIPTION
**Pull request by ghsdh3409**
## Issue
- This pull request is inspired #33 by @sunjae1294.
- This pull request will be merged into #33.

## Changes
- Add `setStreamType()` functionality for Android
- `prepare()` no longer makes new mediaPlayer everytime it is called.

### CODE REVIEW

File: android/src/main/java/com/zmxv/RNSound/RNSoundModule.java [hunk 1]
```diff
@@ -60,61 +58,6 @@ public String getName() {
     return "RNSound";
   }
 
-  // @ReactMethod
```
Comment (ghsdh3409) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java:63]: I removed unnecessary comments.

File: android/src/main/java/com/zmxv/RNSound/RNSoundModule.java [hunk 2]
```diff
@@ -346,12 +289,11 @@ public void stop(final Integer key, final Callback callback) {
   }
 
   @ReactMethod
-  public void reset(final Integer key, final Callback callback) {
```
Comment (ghsdh3409) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java:349]: I removed callback since reset is sync method.

File: index.d.ts [hunk 3]
```diff
@@ -95,7 +95,7 @@ declare class Sound {
    * @param onError Optional callback function if loading file failed
    * @param streamType streamType
    */
-  setStreamType(streamType: { audioStreamType: AudioManagerAudioStreamType }, onError: (error: any) => void)
```
Comment (ghsdh3409) [inline index.d.ts:98]: I think `setStreamType` will be needed only stream type.

File: sound.js [hunk 4]
```diff
@@ -119,36 +119,49 @@ Sound.prototype.stop = function(callback) {
 
 Sound.prototype.setStreamType = function(streamType, callback) {
   // For android only.
-  // resets current media player, sets streamType, and prepare again.
```
Comment (ghsdh3409) [inline sound.js:122]: I removed `reset()` in here! I thought it is more convenient that `setStreamType` includes `reset()`. But, it makes additional logic to send reset result to be needed. Thus, I remove `reset()` and make it raise error when it is called before `reset()`.

**Review by ghsdh3409 — COMMENTED** [review]
Review guide!

**Comment by ghsdh3409** [comment]
@sunjae1294 I refactor your code! Can you review this? This pull request will be merged into your pull request #33.

File: index.d.ts (hunk 3 above)
Comment (ohtangza) [reply index.d.ts:98]: Just out of curiosity, do we have a plan to sync with upstream including this?

Comment (ghsdh3409) [reply index.d.ts:98]: Currently not. It needs more working to sync with upstream.

1) `setAudioStreamType` is deprecated in the API 26.
2) There have been some redundant codes in the `prepare()` when we synced with upstream before.

Let's handle on this other issue.

**Comment by ohtangza** [comment]
@sunjae1294 Please post a comment in the original repo saying that your code has been updated for some reasons after merging it (with the reference and credit to this pull request).

File: android/src/main/java/com/zmxv/RNSound/RNSoundModule.java [hunk 5]
```diff
@@ -60,61 +58,6 @@ public String getName() {
     return "RNSound";
   }
 
-  // @ReactMethod
-  // public void getAudioFocus(final Callback callback) {
```
Comment (sunjae1294) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java:64]: @ghsdh3409 I forgot to remove this commented functions. They are no longer necessary! you can remove them!

File: android/src/main/java/com/zmxv/RNSound/RNSoundModule.java [hunk 6]
```diff
@@ -60,61 +58,6 @@ public String getName() {
...
-  //       } catch(RuntimeException runtimeException) {
-  //         // The callback was already invoked
-  //         Log.e("RNSoundModule", "Exception", runtimeException);
-  //       }
-  //     }
-  //   }, AudioManager.STREAM_MUSIC, AudioManager.AUDIOFOCUS_GAIN_TRANSIENT_EXCLUSIVE);
-  // }
-
-  // @ReactMethod
-  // public void abandonAudioFocus(final Callback callback) {
```
Comment (sunjae1294) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java:92]: @ghsdh3409 Same as above

**Comment by sunjae1294** [comment]
@ghsdh3409 I think the `prepare() no longer makes new mediaPlayer everytime it is called.` part of the summary of this PR should be omitted! I forgot to erase it in my PR.

**Review by sunjae1294 — APPROVED** [review]
Nice!

**Pull request merged**